src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs

```diff
@@ -70,6 +70,9 @@ public class CSharpOperationModel : OperationModelBase<CSharpParameterModel, CSh
                 }
             }
 
+            //TODO: FORK the NJsonSchema.CodeGeneration.CSharp and add own setting for this
+            UseActionResultType = true;
```

Comment:
I think this should be a setting here: https://github.com/RSuter/NSwag/blob/master/src/NSwag.CodeGeneration.CSharp/SwaggerToCSharpControllerGeneratorSettings.cs

Reply:
Done.